### PR TITLE
Shorten `.taskcluster.yml` using mergeDeep

### PR DESCRIPTION
**Pull request by YBCS**
shorten current `.taskcluster.yml` by defining `default_task_definition` and using `$mergeDeep`
closes https://github.com/mozilla/bugbug/issues/268

### CODE REVIEW

**Comment by YBCS** [comment]
@marco-c can you PTAL

File: .taskcluster.yml [hunk 1]
```diff
@@ -33,380 +33,349 @@ tasks:
...
   in:
-    $if: 'tasks_for == "github-push" || (tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "synchronize"])'
-    then:
-      - taskId: { $eval: as_slugid("lint_task") }
+    $let:
+      default_task_definition:
         created: { $fromNow: "" }
         deadline: { $fromNow: "1 hour" }
         provisionerId: proj-bugbug
-        workerType: batch
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml:42]: added now!

File: .taskcluster.yml [hunk 2]
```diff
@@ -33,380 +33,349 @@ tasks:
...
+                    pip install --disable-pip-version-check --no-cache-dir --progress-bar off dist/bugbug-$(cat VERSION).tar.gz &&
+                    pip install --disable-pip-version-check --no-cache-dir --progress-bar off dist/bugbug-$(cat VERSION).tar.gz[nlp] &&
+                    bash <(curl -s https://codecov.io/bash)"
+              metadata:
+                name: bugbug packaging test
+                description: bugbug packaging test
 
-      - taskId: { $eval: as_slugid("docker_build") }
-        created: { $fromNow: "" }
-        deadline: { $fromNow: "3 hours" }
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml:179]: oops! I added it back

File: .taskcluster.yml [hunk 3]
```diff
@@ -33,380 +33,349 @@ tasks:
...
-            - { $eval: as_slugid("version_check_task") }
-            - { $eval: as_slugid("tests_task") }
-            - { $eval: as_slugid("http_tests_task") }
-            - { $eval: as_slugid("frontend_build") }
-            - { $eval: as_slugid("packaging_test_task") }
-          scopes:
-            - secrets:get:project/bugbug/deploy
-          taskId: { $eval: as_slugid("docker_push") }
-          created: { $fromNow: "" }
-          deadline: { $fromNow: "4 hours" }
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml:319]: oops! I added it back

File: .taskcluster.yml [hunk 4]
```diff
@@ -33,380 +33,342 @@ tasks:
...
-            env:
-              TASKCLUSTER_SECRET: project/bugbug/deploy
-              GIT_REPOSITORY: "${repository}"
-              GIT_REVISION: "${head_rev}"
-          metadata:
-            name: bugbug PyPI release
-            description: bugbug PyPI release
-            owner: mcastelluccio@mozilla.com
-            source: ${repository}/raw/${head_rev}/.taskcluster.yml
+        # why does this not have a task id ?
```
Comment (YBCS) [inline .taskcluster.yml:247]: why does this task not have a task id ?

File: .taskcluster.yml [hunk 5]
```diff
@@ -33,380 +33,349 @@ tasks:
...
+                      git -c advice.detachedHead=false checkout ${head_rev} &&
+                      python -c 'import os; print(os.environ.keys())' &&
+                      bash ./scripts/integration_test.sh"
+                  cache:
+                    bugbug-mercurial-repository: /cache
+                metadata:
+                  name: bugbug integration test
+                  description: bugbug integration test
+                  owner: ${user}@users.noreply.github.com
+                  source: ${repository}/raw/${head_rev}/.taskcluster.yml
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml]: removed

File: .taskcluster.yml [hunk 6]
```diff
@@ -33,380 +33,349 @@ tasks:
...
+                  - { $eval: as_slugid("lint_task") }
+                  - { $eval: as_slugid("tests_task") }
+                  - { $eval: as_slugid("http_tests_task") }
+                  - { $eval: as_slugid("frontend_build") }
+                  - { $eval: as_slugid("packaging_test_task") }
+                  - { $eval: as_slugid("version_check_task") }
+                  - { $eval: as_slugid("integration_test") }
+                scopes:
+                  - secrets:get:project/bugbug/deploy
+                provisionerId: proj-bugbug
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml]: removed

File: .taskcluster.yml [hunk 7]
```diff
@@ -33,380 +33,349 @@ tasks:
...
-          name: bugbug ui build
-          description: bugbug ui build
-          owner: mcastelluccio@mozilla.com
-          source: ${repository}/raw/${head_rev}/.taskcluster.yml
+        - $mergeDeep:
+            - { $eval: default_task_definition }
+            - taskId: { $eval: as_slugid("frontend_build") }
+              workerType: batch
+              payload:
+                maxRunTime: 3600
```
[the remark this answers is not in the document]
Comment (YBCS) [reply .taskcluster.yml]: removed

File: .taskcluster.yml (hunk 4 above)
Comment (suhaibmujahid) [reply .taskcluster.yml:247]: Nice catch!

I'm not sure; perhaps it was a mistake. Either way, this is outside the scope of this PR. Could you please create an issue for this and remove the comment from the `.taskcluster.yml` file?

**Review by suhaibmujahid — DISMISSED** [review]
LGTM!

**Comment by marco-c** [comment]
The changes look good to me, we can merge after removing the comment

**Review by marco-c — DISMISSED** [review]

File: .taskcluster.yml (hunk 4 above)
Comment (YBCS) [reply .taskcluster.yml]: comment removed

**Comment by YBCS** [comment]
@marco-c @suhaibmujahid  I have addressed comments PTAL!

**Review by suhaibmujahid — APPROVED** [review]
Thank you, @YBCS!

File: .taskcluster.yml (hunk 4 above)
Comment (suhaibmujahid) [reply .taskcluster.yml]: Thank you!

**Pull request merged**